base/commandLine.cpp

```diff
@@ -1242,25 +1249,25 @@ bool processSettings(Common::String &command, Common::StringMap &settings, Commo
 	// domain (i.e. a target) matching this argument, or alternatively
 	// whether there is a gameid matching that name.
 	if (!command.empty()) {
-		PlainGameDescriptor gd = EngineMan.findGame(command);
-		if (ConfMan.hasGameDomain(command) || gd.gameId) {
-			bool idCameFromCommandLine = false;
+		PlainGameDescriptor gd;
+		const Plugin *plugin = nullptr;
+		if (ConfMan.hasGameDomain(command)) {
+			// Command is a known target
+			ConfMan.setActiveDomain(command);
+		} else if (gd = EngineMan.findGame(command, &plugin), gd.gameId) {
+			// Command is a known game ID
+			ConfMan.setActiveDomain(command);
+
+			ConfMan.set("gameid", gd.gameId);
+			ConfMan.set("engineid", plugin->get<MetaEngine>().getEngineId());
 
 			// WORKAROUND: Fix for bug #1719463: "DETECTOR: Launching
 			// undefined target adds launcher entry"
 			//
 			// We designate gameids which come strictly from command line
 			// so AdvancedDetector will not save config file with invalid
 			// gameid in case target autoupgrade was performed
-			if (!ConfMan.hasGameDomain(command)) {
-				idCameFromCommandLine = true;
-			}
-
-			ConfMan.setActiveDomain(command);
-
-			if (idCameFromCommandLine)
-				ConfMan.set("id_came_from_command_line", "1");
-
+			ConfMan.set("id_came_from_command_line", "1");
```

Comment:
Just FYI, this workaround is still semi-broken (see bug #2788).

Reply:
Since this PR changes the CLI interface to use qualified game names is it within the scope of this PR to fix this in one sweep?

Reply:
I have pushed a fix for bug #2788 to master since it is a simple correction.